testing-plugin/src/main/kotlin/arrow/meta/plugin/testing/TestingLibraryWrapper.kt

```diff
@@ -10,14 +10,14 @@ import java.nio.file.Paths
 
 private const val DEFAULT_FILENAME = "Example.kt"
 
-internal data class CompilationResult(
```

Comment:
I think this change breaks the purpose of our testing library. I'm still trying to figure out what's the need but I'm thinking about 2 options:
* Thinking about an assertion and improving the testing library with that assertion.
* If it's only necessary to get the classes after the compilation, it seems that's not a goal of our testing library. We could add a function for compiling sources with the use of the compiler plugin in this module.

Does it make sense for you, @jansorg , @i-walker ?

Reply:
I propose that we focus on implementing this deliverable and negotiate with @raulraja what the best approach is. I do understand that you value the set of principles in `testing-plugin` and I am certain there is a solution, which works for everyone.
Though for now, we need the time to iterate and fulfill our deadlines.

Reply:
@i-walker CompilationResult should remain internal unless you have a valid reason to make it public as the compiler tests are hiddng that behind asserts. Why do you need this method public? We are attempting to make the DSL cohesive with the rest of the APIs and both compiler and ide should share the design but this model was not intended to be part of the public API

Reply:
In order to provide typeclass resolution tests we need the metadata, that the compiler-plugin produces. Within an IdeEnvironment we can only reference and create files, but without the engine of a compiler. Naturally, we would lean on something which we already use in `Meta` We are in need of a function in `testing-plugin`, where we can utilize a function `(Source) -> (List<Config>) -> CompilationResult` or something isomorphic.

We're only proposing a collaboration between `testing-plugin` and `idea-plugin`, unless we're dropping tests for `SyntheticResolution`. Consequently, we can not guarantee issues that the current `scala-idea-plugin` has e.g.: redlines on Syntheticmembers and other unresolved references we add to the language.

Reply:
What kind of metadata is that and how is it used? We can just add a new assert for that that gives you that metadata or analyzes.it in the interpreter of the test DSL.

Reply:
We need `.class` files and everything generated in the build folder.
We also need the actual files and directories, no interpretation or assertion.

Reply:
The underlying libraries has the property `outputDirectory, which includes some of the files we need. But we're not able to compile it with the current settings.

Reply:
We grab the metadata and copy it into the buildFolder of each IdeTestEnviroment. After that, we can initiate indexing as it would happen if a user would start a gradle build task in Idea.